module-new/src/components/checkbox/checkbox.component.tsx

```diff
@@ -0,0 +1,171 @@
+  customIndeterminateIndicator?: JSX.Element;
+
+  /** (Optional) A custom JSX.Element for the checked indicator. */
+  customIndicator?: JSX.Element;
+
+  /** (Optional) A string representing the direction of the label relative to the checkbox. Default value is 'horizontal'. */
+  direction?: 'vertical' | 'horizontal';
+
+  /** (Optional) A boolean flag to disable the checkbox input. */
+  disabled?: boolean;
+
+  /** (Optional) A React.ReactNode to display a label for the checkbox input. */
+  label?: React.ReactNode;
+
+  /** (Optional) Class name for the label component */
+  labelClassName?: string;
+
+  /** (Optional) A callback function (newValue: TData) => void to handle state when 'checked' is changed. */
+  onCheckedChange?: (newValue: TData) => void;
+
+  /** (Optional) A string to set a custom data-testid attribute for the checkbox container. */
+  testId?: string;
+
+  /** (Optional) Classname for the validation errors */
+  validationErrorsClassName?: string;
+
+  /** (Optional) Can be a string or {key, element} key is necessary for animating in new messages   */
+  validationErrorMessages?: ValidationMessage[];
+
+  /** (Optional) A boolean flag to automatically scroll validation error messages into view. */
+  scrollValidationErrorsIntoView?: boolean;
+}
+
+export const Checkbox = React.forwardRef(
+  <TData extends BindType>(
+    {
+      bind,
+      checkboxClassName,
+      checked,
+      customIndicator,
+      customIndeterminateIndicator,
+      direction,
+      disabled,
+      onCheckedChange,
+      label,
+      labelClassName,
+      scrollValidationErrorsIntoView,
+      testId,
+      validationErrorsClassName,
+      validationErrorMessages,
+      ...nativeProps
+    }: ICheckboxProps<TData>,
+    ref: React.ForwardedRef<HTMLButtonElement>
+  ) => {
+    const id = React.useId();
+
+    const globals = useArmstrongConfig({
+      scrollValidationErrorsIntoView,
+      checkboxCustomIndicator: customIndicator,
+      checkboxCustomIndeterminateIndicator: customIndeterminateIndicator,
+    });
+
+    const [boundValue, setBoundValue, bindConfig] = useBindingState(bind, {
+      value: checked,
+      onChange: onCheckedChange,
+      validationErrorMessages,
+      validationMode: globals.validationMode,
+    });
+
+    const onCheckedChangeInternal = React.useCallback<Required<CheckboxProps>['onCheckedChange']>(
+      newValue => {
+        setBoundValue?.(newValue as TData);
+      },
+      [setBoundValue]
+    );
+
+    const indicator = React.useMemo(() => {
+      switch (boundValue) {
+        case 'indeterminate':
+          return globals.checkboxCustomIndeterminateIndicator;
+        default:
+          return globals.checkboxCustomIndicator;
+      }
+    }, [boundValue, globals.checkboxCustomIndicator, globals.checkboxCustomIndeterminateIndicator]);
+
+    return (
+      <StatusWrapper
+        className={concat(validationErrorsClassName, 'arm-input-base')}
```

Comment:
Should this component receive the same class name as the validation errors? I think it needs its own class?

Reply:
Yes indeed, changed in update